haystack/components/preprocessors/embedding_based_document_splitter.py

```diff
@@ -0,0 +1,431 @@
+        document_embedder=embedder,
+        sentences_per_group=2,      # Group 2 sentences before calculating embeddings
+        percentile=0.95,            # Split when cosine distance exceeds 95th percentile
+        min_length=50,              # Merge splits shorter than 50 characters
+        max_length=1000             # Further split chunks longer than 1000 characters
+    )
+    splitter.warm_up()
+    result = splitter.run(documents=[doc])
+
+    # The result contains a list of Document objects, each representing a semantic chunk
+    # Each split document includes metadata: source_id, split_id, and page_number
+    print(f"Original document split into {len(result['documents'])} chunks")
+    for i, split_doc in enumerate(result['documents']):
+        print(f"Chunk {i}: {split_doc.content[:50]}...")
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        document_embedder: DocumentEmbedder,
+        sentences_per_group: int = 3,
+        percentile: float = 0.95,
+        min_length: int = 50,
+        max_length: int = 1000,
+        language: Language = "en",
+        use_split_rules: bool = True,
+        extend_abbreviations: bool = True,
+    ):
+        """
+        Initialize EmbeddingBasedDocumentSplitter.
+
+        :param document_embedder: The DocumentEmbedder to use for calculating embeddings.
+        :param sentences_per_group: Number of sentences to group together before embedding.
+        :param percentile: Percentile threshold for cosine distance. Distances above this percentile
+            are treated as break points.
+        :param min_length: Minimum length of splits in characters. Splits below this length will be merged.
+        :param max_length: Maximum length of splits in characters. Splits above this length will be recursively split.
+        :param language: Language for sentence tokenization.
+        :param use_split_rules: Whether to use additional split rules for sentence tokenization. Applies additional
+            split rules from SentenceSplitter to the sentence spans.
+        :param extend_abbreviations: If True, the abbreviations used by NLTK's PunktTokenizer are extended by a list
+            of curated abbreviations. Currently supported languages are: en, de.
+            If False, the default abbreviations are used.
+        """
+        self.document_embedder = document_embedder
+        self.sentences_per_group = sentences_per_group
+        self.percentile = percentile
+        self.min_length = min_length
+        self.max_length = max_length
+        self.language = language
+        self.use_split_rules = use_split_rules
+        self.extend_abbreviations = extend_abbreviations
+
+        self._init_validation()
+        self.sentence_splitter: SentenceSplitter | None = None
+        self._is_warmed_up = False
+
+    def _init_validation(self) -> None:
+        """
+        Validates initialization parameters.
+        """
+        if self.sentences_per_group <= 0:
```

Comment:
why not check this in init?

Reply:
It's done on `init()` - check line 105: we call `self._init_validation()`